.github/workflows/publish-jvm.yml

```diff
@@ -15,8 +15,57 @@ env:
   RUSTUP_MAX_RETRIES: 10
 
 jobs:
+  build-linux-artifacts:
+    if: github.repository == 'wireapp/core-crypto'
+    name: Build Linux Artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Setup rust"
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          targets: "x86_64-unknown-linux-gnu"
+      - name: Setup cargo-make
+        uses: davidB/rust-cargo-make@v1
+      - name: Build Artifacts
+        run: |
+          cd crypto-ffi
+          cargo make jvm
+      - name: Upload x86_64-unknown-linux-gnu artifacts
+        uses: actions/upload-artifact@v3
+        with:
+            name: x86_64-unknown-linux-gnu
+            path: target/x86_64-unknown-linux-gnu/release/*.so
+
+  build-darwin-artifacts:
+    if: github.repository == 'wireapp/core-crypto'
+    name: Build Darwin Artifacts
+    runs-on: macos-latest
+    steps:
+      - name: Setup rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          targets: "x86_64-apple-darwin,aarch64-apple-darwin"
+      - name: Setup cargo-make
+        uses: davidB/rust-cargo-make@v1
+      - name: Build Artifacts
+        run: |
+          cd crypto-ffi
+          cargo make jvm
+      - name: Upload x86_64-apple-darwin artifacts
+        uses: actions/upload-artifact@v3
+        with:
+            name: x86_64-apple-darwin
+            path: target/x86_64-apple-darwin/release/*.dylib
+      - name: Upload aarch64-apple-darwin artifacts
+        uses: actions/upload-artifact@v3
+        with:
+            name: aarch64-apple-darwin
+            path: target/aarch64-apple-darwin/release/*.dylib
+
   publish-jvm:
     if: github.repository == 'wireapp/core-crypto'
+    name: Publish JVM Package
+    needs: build-linux-artifacts, build-darwin-artifacts
     runs-on: macos-latest
```

Comment:
Does this job still needs to run on mac-os? If not it would be faster on linux

Reply:
No I think it can run linux

Reply:
Updated!